src/cpu/flexcpu/inflight_inst.hh

```diff
@@ -124,10 +129,16 @@ class InflightInst : public ExecContext,
     // Who depends on me?
     std::vector<std::function<void()>> completionCallbacks;
 
-    // How many dependencies have to be resolved before I can run?
+    std::vector<std::function<void()>> effAddrCalculatedCallbacks;
```

Comment:
This is getting a little complicated with 4 different kinds of callbacks and two counters. It might be a good time to document why all of this is necessary. (This is something that could also go in the paper under the design section, so it's not wasted time to write it.)

Reply:
Yea I could add more comments, but fundamentally, the callbacks are for anything that needs to be called during a state transition, and dependencies are things that have to be satisfied before a state transition.